src/Compilers/CSharp/Test/Semantic/Semantics/NullCheckedParameterTests.cs

```diff
@@ -237,14 +237,20 @@ void M2(__arglist)
         [Fact]
         public void NullCheckedMethodValidationWithOptionalNullParameter()
         {
-            // PROTOTYPE : Make it an error for the default parameter to be set to null with a null-checked value.
             var source = @"
 class C
 {
     void M(string name! = null) { }
 }";
             var comp = CreateCompilation(source);
             comp.VerifyDiagnostics();
+
+            // Warning attached in emit diagnostics.
```

Comment:
>Warning attached in emit diagnostics. [](start = 15, length = 37)

Echoing Andy's feedback, this diagnostic should be produced when the parameter is bound, and the current location in emit which produces this diagnostic should perhaps change to a Debug.Assert (because we shouldn't make it there when this error is produced).